### PR TITLE
Add s390x machine

### PR DESCRIPTION
**Pull request by claucece**
This PR is still WP.

@armfazh , where did you get the sha for the golang image for the other machines?

### CODE REVIEW

**Comment by armfazh** [comment]
> @armfazh , where did you get the sha for the golang image for the other machines?

I used `arm64v8/golang:1.14.0` however, I looked for that image today and has a different sha value.
Maybe we must move to tags rather than shasum.

File: .github/workflows/ci-actions.yml [hunk 1]
```diff
@@ -48,6 +48,22 @@ jobs:
...
+    steps:
+      - uses: actions/checkout@v2
+      - name: Pulling Images
+        run: |
+          docker pull -q multiarch/qemu-user-static
+          docker pull -q s390x/golang@$SHA256
+      - name: Testing
+        run: |
+          docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
+          docker run --rm -v `pwd`:`pwd` -w `pwd` s390x/golang@$SHA256 go test -v ./...
```
Comment (armfazh) [inline .github/workflows/ci-actions.yml:66]: I was investigating what happens with s390x, it seems fetching code causes invalid signature, hence one trick is to vendor dependencies. Include the vendor folder and the following:

```suggestion
          docker run --rm -v `pwd`:`pwd` -w `pwd` s390x/golang@$SHA256 go test -mod=vendor -v ./...
```

**Comment by claucece** [comment]
> I used arm64v8/golang:1.14.0 however, I looked for that image today and has a different sha value.

Yeah, I also saw that. I'll see how it looks with tags ;)

File: .github/workflows/ci-actions.yml (hunk 1 above)
Comment (claucece) [reply .github/workflows/ci-actions.yml:66]: Mmm.. not a big fan of vendoring. I'll investigate tomorrow if there is an alternative.. if not, let's use vendor.

**Comment by armfazh** [comment]
@claucece please take a look on #121. I did lots of changes to make it work.
The bug was raised by s390x because it is big endian. In some parts of the code, it always assumes little endian ordering.

**Comment by claucece** [comment]
Closing this PR as the other was merged.

**Pull request closed without merging**